contracts/ThreeHeapOrdering.sol

```diff
@@ -0,0 +1,306 @@
+// SPDX-License-Identifier: GNU AGPLv3
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts/utils/math/SafeCast.sol";
+
+library ThreeHeapOrdering {
+    struct Account {
+        address id; // The address of the account.
+        uint96 value; // The value of the account.
+    }
+
+    struct HeapArray {
+        Account[] accounts; // All the accounts.
+        uint256 size; // The size of the heap portion of the structure, should be less than accounts length, the rest is an unordered array.
+        mapping(address => uint256) indexes; // A mapping from an address to a index in accounts.
```

Comment:
In `HeapOrdering`, we use `indexOf`, which is a little bit more natural to me. What is the reason why you changed it here ?

Reply:
Actually I changed it recently in `HeapOrdering` from `indexes` to `indexOf` (after @MerlinEgalite's review), and I forgot to also change it in this PR. I like `indexOf` better too, I'll change it